### PR TITLE
Migrate Firebase off ObjC module @import statements

### PR DESCRIPTION
**Pull request by bhamiltoncx**
To resolve build-time incompatibilities with newer versions of clang, we'd like to migrate
Firebase off Objective-C module `@import` statements.

In this PR, I:

1. Change `@import Foo;` system framework imports to `#import <Foo/Foo.h>` for non-example/non-source files
2. Add a new script `scripts/check_no_module_imports.sh` to check new `@import` statements don't show up in non-example/non-source files
3. Invoke the new script from `.travis.yml`

Tested with:

```
% ./scripts/build.sh Firebase iOS
```

### CODE REVIEW

File: FirebaseFunctions.podspec [hunk 1]
```diff
@@ -28,6 +28,9 @@ iOS SDK for Cloud Functions for Firebase.
   s.source_files = 'Functions/FirebaseFunctions/**/*'
   s.public_header_files = 'Functions/FirebaseFunctions/Public/*.h'
 
+  s.frameworks = [
+    'UIKit',
```
Comment (bhamiltoncx) [inline FirebaseFunctions.podspec]: I wasn't sure if this was strictly needed; UIKit is only in example code which I wasn't sure was included in the pod.

Comment (paulb777) [inline FirebaseFunctions.podspec]: Why is this needed? I see a UIKit usage in the Functions Example, but not in the library.

Comment (paulb777) [reply FirebaseFunctions.podspec]: No the Example code is not part of the pod. Please remove. cc: @bklimt

Comment (bhamiltoncx) [reply FirebaseFunctions.podspec]: Got it, thanks for clarifying. Removed.

File: scripts/check_no_module_imports.sh~ [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Copyright 2018 Google
```
Comment (paulb777) [inline scripts/check_no_module_imports.sh~]: Remove this file from PR

File: scripts/check_no_module_imports.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+#   * Sample sources
+#
+options=(
+  -n  # show line numbers
+  -I  # exclude binary files
+  '^@import'
+)
+
+git grep "${options[@]}" \
+    -- ':(exclude,glob)**/Example/**' ':(exclude,glob)**/Sample/**'
```
Comment (paulb777) [inline scripts/check_no_module_imports.sh]: This excludes unit tests as well as Example code.   The tests are in

```
 $ ls -ld Example/*/Tests */Example/Tests
drwxr-xr-x  60 paulbeusterien  eng  1920 Mar 28 13:18 Example/Auth/Tests
drwxr-xr-x  11 paulbeusterien  eng   352 Mar 28 12:15 Example/Core/Tests
drwxr-xr-x   9 paulbeusterien  eng   288 Mar 28 12:16 Example/Database/Tests
drwxr-xr-x  24 paulbeusterien  eng   768 Mar 28 12:16 Example/Messaging/Tests
drwxr-xr-x   5 paulbeusterien  eng   160 Mar 28 12:16 Example/Storage/Tests
drwxr-xr-x  13 paulbeusterien  eng   416 Mar 28 12:16 Firestore/Example/Tests
drwxr-xr-x   6 paulbeusterien  eng   192 Mar 28 12:16 Functions/Example/Tests
```

File: scripts/check_no_module_imports.sh~ (hunk 2 above)
Comment (bhamiltoncx) [reply scripts/check_no_module_imports.sh~]: Done.

File: scripts/check_no_module_imports.sh (hunk 3 above)
Comment (bhamiltoncx) [reply scripts/check_no_module_imports.sh]: Ah, I didn't understand these were the main tests for Firebase.

Updated all test sources files to change `@import` to `#import`.

Updated dependent frameworks in the Xcode project framework link step for the tests.

**Comment by bhamiltoncx** [comment]
Thanks, all fixed.

**Comment by bhamiltoncx** [comment]
Rebase on master.

**Comment by bklimt** [comment]
LGTM

**Comment by bhamiltoncx** [comment]
Groovy, thanks!

**Comment by bhamiltoncx** [comment]
@paulb777 : Fixed, thanks.
